libs/application/ui-components/src/utilities/FileUploadController/index.tsx

```diff
@@ -206,27 +246,42 @@ export const FileUploadController = ({
       },
     })
 
-    // Upload each file.
-    newUploadFiles.forEach(async (f: UploadFile) => {
+    const malwareFiles: UploadFile[] = []
+    const uploadPromises = newUploadFiles.map(async (f) => {
       try {
         const res = await uploadFileFlow(f)
 
+        if(!res.isClean) {
+          malwareFiles.push(f)
+        }
         dispatch({
           type: ActionTypes.UPDATE,
           payload: {
             file: f,
-            status: 'done',
+            status: res.isClean ? 'done' : 'error',
             percent: 100,
             key: res.key,
+            url: res.url
           },
         })
       } catch {
         setUploadError(formatMessage(coreErrorMessages.fileUpload))
       }
     })
+
+    await Promise.all(uploadPromises)
```

Comment:
```suggestion
    await Promise.allSettled(uploadPromises)
```
Consider using `Promise.allSettled()` here because `Promise.all()` fails fast and returns with the first failure but allSettled returns, as you may suspect, when all the promises have been settled.

Reply:
ah, did not know this, perfect